### PR TITLE
StorageCluster: make readability enhancement

### PR DESCRIPTION
**Pull request by raghavendra-talur**
Signed-off-by: Raghavendra Talur <raghavendra.talur@gmail.com>

### CODE REVIEW

File: pkg/controller/storagecluster/placement.go [hunk 1]
```diff
@@ -33,13 +33,6 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 		}
 	}
 
-	topologyMap := sc.Status.NodeTopologies
-	if topologyMap != nil && (component == "mon" || component == "mds") {
-		topologyKey := determineFailureDomain(sc)
-		topologyKey, _ = topologyMap.GetKeyValues(topologyKey)
-		podAffinityTerms := placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-		podAffinityTerms[0].PodAffinityTerm.TopologyKey = topologyKey
-	}
```
Comment (anubhav-here) [inline pkg/controller/storagecluster/placement.go:42]: I have had this thought since a long time now. Good that you brought it up.FYI :  @rohantmp

**Comment by sp98** [comment]
@raghavendra-talur could you please add more information about the changes in the description  or Commit message ?

File: pkg/controller/storagecluster/placement.go [hunk 2]
```diff
@@ -33,13 +33,6 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 		}
 	}
 
-	topologyMap := sc.Status.NodeTopologies
-	if topologyMap != nil && (component == "mon" || component == "mds") {
-		topologyKey := determineFailureDomain(sc)
-		topologyKey, _ = topologyMap.GetKeyValues(topologyKey)
-		podAffinityTerms := placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-		podAffinityTerms[0].PodAffinityTerm.TopologyKey = topologyKey
```
Comment (rohantmp) [inline pkg/controller/storagecluster/placement.go:41]: While this looks like code that does nothing (I thought the same thing when I saw it)
It actually modifies the antiaffinity of `placement` before returning it.
```
placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution[0].PodAffinityTerm.TopologyKey = topologyKey
```

Comment (raghavendra-talur) [reply pkg/controller/storagecluster/placement.go:41]: Thanks, I verified it. Here is the playground link for anyone interested https://play.golang.org/p/MEnYgrnX_jr

Comment (raghavendra-talur) [reply pkg/controller/storagecluster/placement.go:41]: To elaborate more on previous comment: When you copy a struct in golang, you make a shallow copy. Only those members of the struct that are of the basic data type get copied by value and the rest get copied by reference. On line 40 above, a slice that is a member of the inner struct of the placement is being copied to podAffinityTerms and therefore is copied by reference.

Comment (raghavendra-talur) [reply pkg/controller/storagecluster/placement.go:41]: Thanks! Yes, that is correct.

File: pkg/controller/storagecluster/placement.go [hunk 3]
```diff
@@ -33,13 +33,6 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 		}
 	}
 
-	topologyMap := sc.Status.NodeTopologies
-	if topologyMap != nil && (component == "mon" || component == "mds") {
-		topologyKey := determineFailureDomain(sc)
-		topologyKey, _ = topologyMap.GetKeyValues(topologyKey)
```
Comment (raghavendra-talur) [inline pkg/controller/storagecluster/placement.go:39]: @jarrpa Once we determine the topologyKey for the StorageCluster, what is the reason to get the keyvalue pairs but discard the values?

Is is possible that the topologyKey obtained using determineFailureDomain function is a substring of the "real" key?

**Comment by raghavendra-talur** [comment]
@rohantmp  @jarrpa @anubhav-here I have rebased on master now. If you believe this change is not worth it, I am willing to close the PR.

**Review by jarrpa — APPROVED** [review]
Eh, I disagree that this is easier to read, but that's probably because I wrote it. :) LGTM

**Pull request merged**